parser/block_test.go

```diff
@@ -19,68 +19,122 @@ import (
 )
 
 func TestBlockParser(t *testing.T) {
-	// These (valid on testnet) correspond to the transactions in testdata/blocks
-	var txhashes = []string{
-		"81096ff101a4f01d25ffd34a446bee4368bd46c233a59ac0faf101e1861c6b22",
-		"921dc41bef3a0d887c615abac60a29979efc8b4bbd3d887caeb6bb93501bde8e",
-		"d8e4c336ffa69dacaa4e0b4eaf8e3ae46897f1930a573c10b53837a03318c980",
-		"4d5ccbfc6984680c481ff5ce145b8a93d59dfea90c150dfa45c938ab076ee5b2",
-		"df2b03619d441ce3d347e9278d87618e975079d0e235dfb3b3d8271510f707aa",
-		"8d2593edfc328fa637b4ac91c7d569ee922bb9a6fda7cea230e92deb3ae4b634",
+	// These (valid on testnet) correspond to the transactions in testdata/blocks;
+	// for each block, the hashes for the tx within that block.
+	var txhashes = [][]string{
+		{
+			"81096ff101a4f01d25ffd34a446bee4368bd46c233a59ac0faf101e1861c6b22",
+		}, {
+			"921dc41bef3a0d887c615abac60a29979efc8b4bbd3d887caeb6bb93501bde8e",
+		}, {
+			"d8e4c336ffa69dacaa4e0b4eaf8e3ae46897f1930a573c10b53837a03318c980",
+			"4d5ccbfc6984680c481ff5ce145b8a93d59dfea90c150dfa45c938ab076ee5b2",
+		}, {
+			"df2b03619d441ce3d347e9278d87618e975079d0e235dfb3b3d8271510f707aa",
+			"8d2593edfc328fa637b4ac91c7d569ee922bb9a6fda7cea230e92deb3ae4b634",
+		},
```

Comment:
This is simpler than the old code; the old code "knew" that there were 4 blocks containing 1, 1, 2, 2 transactions respectively. So looping over the blocks, and for each block looping over its transactions, happens to amount to exactly 6 transactions  The new code specifically lists the txids in each block. The reason for this change is not just to simplify, it's that now the blocks will gain more transactions as the test runs (repeats of the one or 2 that it already has, see below).